src/frontend/apps/desk/src/features/mail-domains/components/forms/CreateMailboxForm.tsx

```diff
@@ -0,0 +1,292 @@
+
+  const methods = useForm<CreateMailboxParams>({
+    delayError: 0,
+    defaultValues: {
+      first_name: '',
+      last_name: '',
+      local_part: '',
+      secondary_email: '',
+      phone_number: '',
+    },
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    resolver: zodResolver(createMailboxValidationSchema),
+  });
+
+  const { mutate: createMailbox, ...queryState } = useCreateMailbox({
+    domainId: mailDomain.id,
+    onSuccess: async () => {
+      await queryClient.prefetchQuery({
+        queryKey: [KEY_LIST_MAILBOX, { id: mailDomain.id }],
+      });
+
+      toast(t('Mailbox created!'), VariantType.SUCCESS, {
+        duration: 4000,
+      });
+
+      setIsFormVisible(false);
+    },
+  });
+
+  const closeModal = () => setIsFormVisible(false);
+
+  const onSubmitCallback = (event: React.FormEvent) => {
+    event.preventDefault();
+    void methods.handleSubmit((data) =>
+      createMailbox({ ...data, mailDomainId: mailDomain.id }),
+    )();
+  };
+
+  return (
+    <FormProvider {...methods}>
+      <Modal
+        isOpen
+        leftActions={
+          <Button
+            color="secondary"
+            fullWidth
+            onClick={closeModal}
+            disabled={methods.formState.isSubmitting}
+          >
+            {t('Cancel')}
+          </Button>
+        }
+        onClose={closeModal}
+        closeOnClickOutside
+        hideCloseButton
+        rightActions={
+          <Button
+            color="primary"
+            fullWidth
+            type="submit"
+            form={FORM_ID}
+            disabled={methods.formState.isSubmitting}
+          >
+            {t('Submit')}
+          </Button>
+        }
+        size={ModalSize.MEDIUM}
+        title={
+          <Box $align="center" $gap="1rem">
+            <IconCreateMailbox
+              width={48}
+              color={colorsTokens()['primary-text']}
+              title={t('Mailbox creation form')}
+            />
+            <Text $size="h3" $margin="none" role="heading" aria-level={3}>
+              {t('Create a mailbox')}
+            </Text>
+          </Box>
+        }
+      >
+        <Box $width="100%" $margin={{ top: 'large', bottom: 'xl' }}>
+          {queryState.isError && (
+            <TextErrors className="mb-s" causes={queryState.error.cause} />
+          )}
+          {methods ? (
+            <Form
+              methods={methods}
+              t={t}
+              mailDomain={mailDomain}
+              onSubmitCallback={onSubmitCallback}
+            />
+          ) : null}
+        </Box>
+      </Modal>
+    </FormProvider>
+  );
+};
+
+const Form = ({
+  methods,
+  t,
+  mailDomain,
+  onSubmitCallback,
+}: {
+  methods: UseFormReturn<CreateMailboxParams>;
+  t: TFunction;
+  mailDomain: MailDomain;
+  onSubmitCallback: (event: React.FormEvent) => void;
+}) => (
+  <form onSubmit={onSubmitCallback} id={FORM_ID}>
+    <Box $direction="column" $width="100%" $gap="3rem" $margin="auto">
```

Comment:
Seems to fit more to the design.
```suggestion
    <Box $direction="column" $width="100%" $gap="2rem" $margin="auto">
```